### PR TITLE
Add the session idle timeout fonctionnality

### PR DESCRIPTION
**Pull request by gurtzoo**
The Idle Session Timeout (Inactivity) represents the amount of time a user can be inactive before the user's session times out in seconds.
The Idle Timeout is working with the session length(lifetime) to managed the amount of time the user session cookie­ expired.
This parameter is disabled by default to maintain backward compatibility.

### CODE REVIEW

File: auth_mellon_handler.c [hunk 1]
```diff
@@ -1629,7 +1629,7 @@ static int am_validate_unique_assertion_id(request_rec *r,
  *  Nothing.
  */
 static void am_handle_session_expire(request_rec *r, am_cache_entry_t *session,
-                                LassoSaml2Assertion *assertion)
+                                     LassoSaml2Assertion *assertion)
```
Comment (simo5) [inline auth_mellon_handler.c:1632]: Can you please avoid unrelated changes here ?

File: auth_mellon_handler.c [hunk 2]
```diff
@@ -1673,6 +1673,8 @@ static void am_handle_session_expire(request_rec *r, am_cache_entry_t *session,
     }
 }
 
+
+
```
Comment (simo5) [inline auth_mellon_handler.c:1677]: and here as well ...

File: auth_mellon_cache.c [hunk 3]
```diff
@@ -465,6 +467,25 @@ void am_cache_update_expires(request_rec *r, am_cache_entry_t *t, apr_time_t exp
...
+ *  am_cache_entry_t *t      The current session.
+ *  apr_time_t idle_timeout  The new timestamp.
+ *
+ * Returns:
+ *  Nothing.
+ */
+void am_cache_update_idle_timeout(request_rec *r, am_cache_entry_t *t, apr_time_t idle_timeout)
+{
+    /* If idle_timeout timestamp is not -1 (disabled), update the value. */
+    if(t->idle_timeout != -1) {
```
Comment (simo5) [inline auth_mellon_cache.c:484]: Why this check here ?
There are checks about whether the default is -1 in the code that invoke this function.

Maybe either move the logic all in here, or just set the timeout detemined by the caller unconditionally?

It is a bit confusing to have the logic split across.

**Review by simo5 — CHANGES_REQUESTED** [review]
In general this patch looks good to me.

A few nits about spurious changes.

Additionally I see quit a few long lines.
I do not recall if we have a strict rule on line length in this project but I strongly prefer to stay within 80 columns for readability.

File: auth_mellon_handler.c (hunk 1 above)
Comment (gurtzoo) [reply auth_mellon_handler.c:1632]: ok

File: auth_mellon_handler.c (hunk 2 above)
Comment (gurtzoo) [reply auth_mellon_handler.c:1677]: ok

File: auth_mellon_cache.c (hunk 3 above)
Comment (gurtzoo) [reply auth_mellon_cache.c:484]: understand

File: auth_mellon_handler.c (hunk 1 above)
Comment (gurtzoo) [reply auth_mellon_handler.c]: done

File: auth_mellon_handler.c (hunk 2 above)
Comment (gurtzoo) [reply auth_mellon_handler.c]: done

File: auth_mellon_cache.c (hunk 3 above)
Comment (gurtzoo) [reply auth_mellon_cache.c]: done, i have decided to centralize the logic here.

**Comment by gurtzoo** [comment]
modifications and squash done.

**Comment by simo5** [comment]
Sounds like you forgot to force push the squashed commit, I still see the original commit + a fixup one.

**Comment by gurtzoo** [comment]
Sorry, now should be pushed.

**Comment by simo5** [comment]
I see now 4 commits, one of which is a merge commit.

Please do this in the gurtzoo:sesion-idle branch:

$ git rebase -i origin/master
[in the editor squash all commits into one]
$ git push -f <remote-tree-id> session-idle

**Comment by simo5** [comment]
Alternatively click the checkmark that allows me to edit your branch and I'll fix it for you.

**Comment by gurtzoo** [comment]
> Alternatively click the checkmark that allows me to edit your branch and I'll fix it for you.

Yes i have check the box, you can correct it. I have tried on my side but, again, problem with git.

**Comment by simo5** [comment]
Fixed

**Review by simo5 — APPROVED** [review]

**Comment by gurtzoo** [comment]
good, thank you.

**Comment by gurtzoo** [comment]
Do i need to do something to be merge in master?

**Review by jhrozek — APPROVED** [review]
lgtm, too.

**Comment by jhrozek** [comment]
> Do i need to do something to be merge in master?

Thank you for the contribution. The PR looks good to me as well, so I'm going to merge it.
